### PR TITLE
docs: add cli README

### PR DESCRIPTION
**Pull request by zoumo**
**What this PR does**
add readme for cli

**why we need it**

**Special notes for your reviewer**:

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Cli - Mamba
+
+The CLI pkg also named Mamba.
```
Comment (Jimexist) [inline cli/README.md]: is

File: cli/README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+## About the project
+
+Mamba provide a higher level and more friendly interfaces to build modern command line interfaces and manage configurations for Go applications.
```
Comment (Jimexist) [inline cli/README.md]: provides

File: cli/README.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+It bases on
```
Comment (Jimexist) [inline cli/README.md]: is based

File: cli/README.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+-   [spf13/pflag](https://github.com/spf13/pflag): to provide POSIX/GNU-style flags
+-   [spf13/cobra](https://github.com/spf13/cobra): to provide a simple interface to create powerful modern CLI interfaces
+-   [spf13/viper](https://github.com/spf13/viper): to provide a complete configuration sulution for [12-Factor](https://12factor.net/zh_cn/) apps
```
Comment (Jimexist) [inline cli/README.md]: remove `zh_cn` postfix

File: cli/README.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+## Status
+
+**Working in process**
+
+## Concepts
+
+A cli is composed of commands, argumetns & flags.
```
Comment (Jimexist) [inline cli/README.md]: arguments

Comment (Jimexist) [reply cli/README.md]: and

File: cli/README.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+**Commands** represent actions, **Args** are things and **Flags** are modifiers for those actions.
```
Comment (Jimexist) [inline cli/README.md]: things?

File: cli/README.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+The pattern to follow is `appname commands args --flag`
+
+The flag is compatible with the [GNU extensions to the POSIX recommendations for command-line options](http://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html).
```
Comment (Jimexist) [inline cli/README.md]: the flags follow the style mentioned in... ?

File: cli/README.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+## Getting Started
+
+### Commands
+
+You can migrate your cobra commands to mamba easily.
+
+the `cli.NewCommand` receive a `*cobra.Command` to build the whole CLI.
```
Comment (Jimexist) [inline cli/README.md]: receives

Comment (Jimexist) [reply cli/README.md]: the whole? what do you mean?

File: cli/README.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+		},
+	})
+}
+```
+
+[More about cobra command](https://godoc.org/github.com/spf13/cobra#Command)
+
+### Flags
+
+Flag, different from `pflag.Flag`, is an interface implementing the following functions in Mamba.
```
Comment (Jimexist) [inline cli/README.md]: interface is to be implemented, not implementing other functions right?

File: cli/README.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+
+And it is declarative in Mamba not imperative like Cobra. That makes the flags **more readable**.
```
Comment (Jimexist) [inline cli/README.md]: , unlike in Cobra where it is imperative, which makes...

File: cli/README.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+Mamba will bind `viper` and `flags` automatically. That means you can get all your flag value from a configuration center.
```
Comment (Jimexist) [inline cli/README.md]: values

Comment (Jimexist) [reply cli/README.md]: what is a configuration center?

File: cli/README.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+	// if EnvKey is set, it will override the automatic generated env key
+	EnvKey string
+	// the default value
+	DefValue string
+	// points to a variable in which to store the value of the flag
+	Destination *string
+}
+```
+
+#### Bind flag with ENV
```
Comment (Jimexist) [inline cli/README.md]: Binding

File: cli/README.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+Mamba supports the ability to bind your flags with env.
```
Comment (Jimexist) [inline cli/README.md]: env => envvar or environment variables

File: cli/README.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+Mamba use the following precedence order. Each item takes precedence over the item bellow it.
```
Comment (Jimexist) [inline cli/README.md]: uses

Comment (Jimexist) [reply cli/README.md]: below

Comment (Jimexist) [reply cli/README.md]: > Mamba looks for the value for flags in this order:

this shall suffice

File: cli/README.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+
+-   explicit call to set
```
Comment (Jimexist) [inline cli/README.md]: calls

File: cli/README.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+-   flag
+-   env
+-   default
```
Comment (Jimexist) [inline cli/README.md]: default values

File: cli/README.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+
+Mamba supports the ability to bind you flags with ENV and add prefix automatically
+
+The following functions exist to aid working with ENV
+
+-   `AutomaticEnv()`
+-   `SetEnvPrefix(string)`
+-   `SetEnvKeyReplacer(*strings.Replacer)`
+
+*When working with ENV variables, it’s important to recognize that Mamba treats ENV variables as case insensitive. All ENV variables are treated as UPPER case*
```
Comment (Jimexist) [inline cli/README.md]: treats environment variables case insensitively.

File: cli/README.md [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+
+By using `AutomaticEnv` , you tell mamba to bind all flags with ENV automatically.
```
Comment (Jimexist) [inline cli/README.md]: remove space before `,`

File: cli/README.md [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+-   `SetEnvPrefix(string)`
+-   `SetEnvKeyReplacer(*strings.Replacer)`
+
+*When working with ENV variables, it’s important to recognize that Mamba treats ENV variables as case insensitive. All ENV variables are treated as UPPER case*
+
+By using `AutomaticEnv` , you tell mamba to bind all flags with ENV automatically. 
+
+`SetEnvPrefix` is **always** working with `AutomaticEnv` . It makes mamba add a prefix while reading from env variables.
+
+By using  `SetEnvKeyReplacer` , you make mamba change the ENV by key replacer. For example, you can set a `UnderlineReplacer` to replace all `-` with `_` .
```
Comment (Jimexist) [inline cli/README.md]: remove extra spaces

Comment (Jimexist) [reply cli/README.md]: an

File: cli/README.md [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+os.Setenv("NIRVANA_LOG", "TEST")
+
+cmd.AddFlag(f)
+
+// *log is TEST
+```
+
+#### Hidden,Deprecated, ShorthandDeprecated
```
Comment (Jimexist) [inline cli/README.md]: space after `,`

File: cli/README.md (hunk 6 above)
Comment (zoumo) [reply cli/README.md]: this sentence is copied from from cobra

File: cli/README.md (hunk 7 above)
Comment (zoumo) [reply cli/README.md]: copy from pflag

**Pull request merged**